### PR TITLE
support vector field

### PR DESCRIPTION
**Pull request by wd0517**
Now TiDB serverless support vector data type, this pr let django-tidb support VectorField to communite with it.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -133,6 +137,56 @@ But there are some limitations:
...
+
+- `L1Distance`
+- `L2Distance`
+- `CosineDistance`
+- `NegativeInnerProduct`
+
+Get instances with vector field and calculate distance to a given vector:
+
+```python
+
```
Comment (dveeden) [inline README.md:180]:
```suggestion
```

**Review by dveeden — COMMENTED** [review]
Should we wait for this to be in public beta and/or pingcap/tidb master?

File: django_tidb/fields/vector.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    if value is None or isinstance(value, np.ndarray):
+        return value
+
+    if isinstance(value, bytes):
+        value = value.decode("utf-8")
+
+    return np.array(value[1:-1].split(","), dtype=np.float32)
+
+
+class VectorField(Field):
```
Comment (dveeden) [inline django_tidb/fields/vector.py:41]:
```suggestion
class VectorField(Field):
"""
Support for AI Vector storage.

Status: Beta

Info: https://www.pingcap.com/blog/integrating-vector-search-into-tidb-for-ai-applications/
"""
```

Maybe copy some more of the README into code comments and extend them

**Review by dveeden — COMMENTED** [review]
Is there any testing/CI for this?

**Comment by wd0517** [comment]
> Should we wait for this to be in public beta and/or pingcap/tidb master?

As far as I know, PingCAP will soon release the vector data type in TiDB Serverless. We should prepare the SDK beforehand and do not need to wait for public beta. Additionally, we have already marked VectorField as beta in the document.

**Comment by wd0517** [comment]
> Is there any testing/CI for this?

There are only manual test for now, since it can only test in TiDB Serverless, its not suitable for current test script.

**Comment by wd0517** [comment]
> Treat it as a variant or default? Maybe just like some features are only available on some specific versions? Sooner or later I think the vector feature should be available in all TiDB deployments.

It will only works in TiDB Cloud Serverless Cluster as far as I know, I haven't received a clear plan regarding when it will be available in pingcap/tidb master.

The `VectorField` is defined and enabled by the user, not as a built-in feature. Therefore, there is no need to check the TiDB version in django-tidb.

**Comment by wd0517** [comment]
Add tests: https://github.com/pingcap/django-tidb/pull/60/commits/46f73a0372af08c6e5309db556363e97952fb301

There is currently only one test cluster available, so it does not support running tests in parallel. I will utilize the TiDB Serverless branch feature once vector is released to TiDB Cloud Production.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -80,3 +80,47 @@ jobs:
...
+          python -m pip install tox tox-gh-actions
+          sudo apt-get update
+          sudo apt-get install -y libmemcached-dev zlib1g-dev
+
+      - name: Hack for vector tests
+        run: |
+          sed '27a cp -r ./tests/tidb_vector/ $DJANGO_TESTS_DIR/django/tests/tidb_vector/' -i django_test_suite.sh
+          sed '31a pip install numpy~=1.0' -i django_test_suite.sh
+          cat django_test_suite.sh
+          echo "tidb_vector" > django_test_apps.txt
```
Comment (zhangyangyu) [inline .github/workflows/ci.yml:120]: Seems there is a trailing newline?
image
`echo -n` maybe?

**Review by zhangyangyu — APPROVED** [review]

**Pull request merged**